sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/SynchronousMessageSubscriber.java

```diff
@@ -272,35 +277,12 @@ private SynchronousReceiveWork getOrUpdateCurrentWork() {
             }
 
             currentWork = workQueue.poll();
-            while (currentWork != null) {
```

Comment:
I remove this code here for two reason from my understanding:
1. the work in queue is never started, so it won't be terminated except the subscriber disposed.
2. the work in queue won't send request, so we if add the remaining to current REQUESTED, it may be wrong. The difference has been calculate when try to requestUpStream().

Reply:
Thinking a bit more, currently the work timeout start at the time when work start.  which means if 2 works to process, for the 2nd work, receiveMessage() timeout = 1st work timeout + 2nd work timeout.
Should the work timeout start at the time we put the work into queue?

Reply:
This seems depends on the design, on multiple thread call the iterator. And whether the timeout is for network (that message requested but not able to receive), or for user (that stream called but not able to progress, maybe because a prior thread still receiving a few million message?)

Reply:
Yes, this was a design choice made. The typical application use case is "obtain_one_iterable -> iterate -> iteration_completed", then repeat,  the cycle is serial.

Even if the application obtains multiple "Iterables", it can iterate only one "Iterable" at a time, which we call "Active-Iterable".  The current "Active-Iterable" has to terminate for the next "Iterable" already in the scope to become "Active-Iterable". So the design was to start the timer only when an Iterable transition to "Active-Iterable"